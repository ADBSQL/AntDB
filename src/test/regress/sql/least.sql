set grammar to oracle;
select least(1,2,3,4) from dual;
select least(1.1,to_number(1.1),exp(2)) from dual;
select least(5.2,to_number(1.1),to_char(1.1)) from dual;
select least(5.2,to_number(1.1),'1') from dual;
select least(to_char(100),to_char(12),to_char(3.1)) from dual;
select least(to_number(3),'100','12') from dual;
select least('100',to_number(3),12) from dual;
select least('100','12',to_number(3)) from dual;
select least('100','12',exp(2)) from dual;
select least('abc','ab','abcd') from dual;
select least('abc','ab','1') from dual;
--select least('你好','加油') from dual;
select least('','a') from dual;
select least('a',null) from dual;
select least(null,'') from dual;
