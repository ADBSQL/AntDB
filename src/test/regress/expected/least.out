set grammar to oracle;
select least(1,2,3,4) from dual;
 least 
-------
     1
(1 row)

select least(1.1,to_number(1.1),exp(2)) from dual;
 least 
-------
   1.1
(1 row)

select least(5.2,to_number(1.1),to_char(1.1)) from dual;
 least 
-------
   1.1
(1 row)

select least(5.2,to_number(1.1),'1') from dual;
 least 
-------
     1
(1 row)

select least(to_char(100),to_char(12),to_char(3.1)) from dual;
 least 
-------
 100
(1 row)

select least(to_number(3),'100','12') from dual;
 least 
-------
     3
(1 row)

select least('100',to_number(3),12) from dual;
 least 
-------
 100
(1 row)

select least('100','12',to_number(3)) from dual;
 least 
-------
 100
(1 row)

select least('100','12',exp(2)) from dual;
 least 
-------
 100
(1 row)

select least('abc','ab','abcd') from dual;
 least 
-------
 ab
(1 row)

select least('abc','ab','1') from dual;
 least 
-------
 1
(1 row)

--select least('你好','加油') from dual;
select least('','a') from dual;
 least 
-------
 
(1 row)

select least('a',null) from dual;
 least 
-------
 
(1 row)

select least(null,'') from dual;
 least 
-------
 
(1 row)

